File: api/ECommerce.Api/Models/Subcategory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ECommerce.Api.Models
{
    public class Subcategory
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        [MaxLength(2048)]
        public string? Image { get; set; }

        [Required]
        [MaxLength(100)]
        public string CategoryId { get; set; } = string.Empty;

        public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAtUtc { get; set; }

        public bool IsActive { get; set; } = true;

        public int SortOrder { get; set; } = 0;

        // Navigation properties
        [ForeignKey("CategoryId")]
        public virtual Category? Category { get; set; }

        public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: sportapp/src/types/subcategory.ts
export interface Subcategory {
  id: string;
  name: string;
  description: string;
  image?: string;
  categoryId: string;
  createdAtUtc?: string;
  updatedAtUtc?: string;
  isActive: boolean;
  sortOrder: number;
  categoryName?: string;
  productCount?: number;
}

export interface CreateSubcategoryRequest {
  name: string;
  description: string;
  image?: string;
  isActive: boolean;
  sortOrder: number;
}

export interface UpdateSubcategoryRequest {
  name: string;
  description: string;
  image?: string;
  isActive: boolean;
  sortOrder: number;
}
